scripts/storage/qcow2/managesnapshot.py

```diff
@@ -0,0 +1,279 @@
+        logging.error('%s', e)
+        raise
+
+    # Removing old snapshot file
+    try:
+        logging.info('Removing old snapshot file %s', snapshot_file_path)
+        os.remove(snapshot_file_path)
+        logging.info('Successfully removed snapshot file %s', snapshot_file_path)
+    except Exception as e:
+        logging.error('An error occured during the removal of snapshot file %s', snapshot_file_path)
+        logging.error('%s', e)
+        raise
+
+
+def backup_snapshot():
+    src = disk_path
+    dst = backupdir + "/" + backup_file_name
+    logging.info('Backup snapshot request for snapshot %s on domain %s', src, domain)
+
+    try:
+        logging.info('Backuping snapshot %s to %s', src, dst)
+        if not os.path.isdir(backupdir):
+            os.makedirs(backupdir, 0777)
+        shutil.copyfile(src, dst)
+        logging.info('Snapshot backup completed %s to %s', src, dst)
+    except Exception as e:
+        logging.error('An error occured during snapshot backup')
+        logging.error('%s', e)
+        logging.error('Destroying snapshot')
+        destroy_snapshot()
+        raise
+
+
+def getdev(dom0, snapshot_file_path):
+    xmldesc = dom0.XMLDesc(0)
+    root = etree.fromstring(xmldesc)
+    for i in root.findall('./devices/disk'):
+        device = {}
+        if 'disk' in i.attrib.values():
+            child = i.getchildren()
+            for childitems in child:
+                if childitems.tag == 'source':
+                    device['source'] = (childitems.get('file'))
+                elif childitems.tag == 'target':
+                    device['dev'] = (childitems.get('dev'))
+        if snapshot_file_path in device.values():
+            dev = device.get('dev')
+            return dev
+
+
+def wait_for_block_job(dom0, dev, abort_on_error=False,
+                       wait_for_job_clean=False):
+        """Wait for libvirt block job to complete.
+        Libvirt may return either cur==end or an empty dict when
+        the job is complete, depending on whether the job has been
+        cleaned up by libvirt yet, or not.
+        :returns: True if still in progress
+                  False if completed
+        """
+
+        status = dom0.blockJobInfo(dev, 0)
+        if status == -1 and abort_on_error:
+            logging.error('libvirt error while requesting blockjob info')
+            raise libvirt.VIR_ERR_OPERATION_FAILED('libvirt error while requesting blockjob info')
+        try:
+            cur = status.get('cur', 0)
+            end = status.get('end', 0)
+        except Exception:
+            return False
+
+        if wait_for_job_clean:
+            job_ended = not status
+        else:
+            job_ended = cur == end
+
+        return not job_ended
+
+
+if __name__ == "__main__":
+    global createarg, deletearg, backuparg, backupdir, backup_file_name, disk_path, domain, snapshot_file_name, snapshot_file_path
```

Comment:
A couple of these variables don't need to be global, and for the rest it'd be more robust to pass them as arguments to the relevant functions that need them. Suggested change:

```diff
diff --git a/scripts/storage/qcow2/managesnapshot.py b/scripts/storage/qcow2/managesnapshot.py
index 66227b6..1602575 100755
--- a/scripts/storage/qcow2/managesnapshot.py
+++ b/scripts/storage/qcow2/managesnapshot.py
@@ -33,7 +33,7 @@ def getargs():
     return args
 
 
-def create_snapshot():
+def create_snapshot(disk_path, domain, snapshot_file_path):
 
     logging.info('Create snapshot request for disk %s on VM %s', disk_path, domain)
 
@@ -87,11 +87,11 @@ def create_snapshot():
         logging.error('Failed to create snapshot %s for vm %s', snapshot_file_path, domain)
         logging.error('%s', e)
         logging.error('Destroying snapshot')
-        destroy_snapshot()
+        destroy_snapshot(disk_path, domain, snapshot_file_path)
         raise
 
 
-def destroy_snapshot():
+def destroy_snapshot(disk_path, domain, snapshot_file_path):
 
     logging.info('Destroy snapshot request for disk %s on VM %s', snapshot_file_path, domain)
 
@@ -156,7 +156,7 @@ def destroy_snapshot():
         raise
 
 
-def backup_snapshot():
+def backup_snapshot(disk_path, domain, backupdir, backup_file_name, snapshot_file_path):
     src = disk_path
     dst = backupdir + "/" + backup_file_name
     logging.info('Backup snapshot request for snapshot %s on domain %s', src, domain)
@@ -171,7 +171,7 @@ def backup_snapshot():
         logging.error('An error occured during snapshot backup')
         logging.error('%s', e)
         logging.error('Destroying snapshot')
-        destroy_snapshot()
+        destroy_snapshot(disk_path, domain, snapshot_file_path)
         raise
 
 
@@ -221,7 +221,6 @@ def wait_for_block_job(dom0, dev, abort_on_error=False,
 
 
 if __name__ == "__main__":
-    global createarg, deletearg, backuparg, backupdir, backup_file_name, disk_path, domain, snapshot_file_name, snapshot_file_path
     args = getargs()
     createarg = args['createarg']
     destroyarg = args['destroyarg']
@@ -250,11 +249,11 @@ if __name__ == "__main__":
 
     try:
         if createarg:
-            create_snapshot()
+            create_snapshot(disk_path, domain, snapshot_file_path)
         elif destroyarg:
-            destroy_snapshot()
+            destroy_snapshot(disk_path, domain, snapshot_file_path)
         elif backuparg:
-            backup_snapshot()
+            backup_snapshot(disk_path, domain, backupdir, backup_file_name, snapshot_file_path)
 
         # No error, we send an ok riemann event
         service = "Cloudstack-Snapshot-%s" % (domain)
```